src/HealthChecks.Publisher.Seq/SeqPublisher.cs

```diff
@@ -54,29 +57,50 @@ public async Task PublishAsync(HealthReport report, CancellationToken cancellati
                 }
             };
 
-            await PushMetricsAsync(JsonConvert.SerializeObject(events));
+            await PushMetricsAsync(JsonConvert.SerializeObject(events), cancellationToken);
         }
 
-        private async Task PushMetricsAsync(string json)
+        private async Task PushMetricsAsync(string json, CancellationToken cancellationToken)
         {
             try
             {
-                var httpClient = _httpClientFactory();
+                using var httpClient = _httpClientFactory();
 
-                using var pushMessage = new HttpRequestMessage(HttpMethod.Post, $"{_options.Endpoint}/api/events/raw?apiKey={_options.ApiKey}")
+                using var pushMessage = new HttpRequestMessage(HttpMethod.Post, _checkUri)
                 {
                     Content = new StringContent(json, Encoding.UTF8, "application/json")
                 };
 
-                using var response = await httpClient.SendAsync(pushMessage, HttpCompletionOption.ResponseHeadersRead);
+                using var response = await httpClient.SendAsync(
+                    pushMessage,
+                    HttpCompletionOption.ResponseHeadersRead,
+                    cancellationToken);
+
                 response.EnsureSuccessStatusCode();
             }
             catch (Exception ex)
             {
-                Trace.WriteLine($"Exception is throwed publishing metrics to Seq with message: {ex.Message}");
+                Trace.WriteLine($"Exception thrown publishing metrics to Seq with message: {ex.Message}");
             }
         }
 
+        private static Uri BuildCheckUri(SeqOptions options)
+        {
+            if (string.IsNullOrEmpty(options.Endpoint))
+                throw new ArgumentNullException(nameof(options.Endpoint));
+
+            var uriBuilder = new UriBuilder(options.Endpoint)
+            {
+                Path = "/api/events/raw",
+            };
+
+            // Add api key if supplied
+            if (!string.IsNullOrWhiteSpace(options.ApiKey))
```

Comment:
```suggestion
            if (!string.IsNullOrEmpty(options.ApiKey))
```

or just `if (options.ApiKey != null)`

Reply:
Ready to merge after this fix.